src/Discord/Parts/Guild/Guild.php

```diff
@@ -594,6 +594,49 @@ public function updateRolePositions(array $roles): ExtendedPromiseInterface
             });
     }
 
+    /**
+     * Returns a list of guild member objects whose username or nickname starts with a provided string.
+     *
+     * @param array $options An array of options.
+     *                       query => query string to match username(s) and nickname(s) against
+     *                       limit => how many entries are returned (default 1, minimum 1, maximum 1000)
+     *
+     * @return ExtendedPromiseInterface
+     */
+    public function searchMembers(array $options): ExtendedPromiseInterface
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefined([
+            'query',
+            'limit',
+        ])
+        ->setDefaults(['limit' => 1])
+        ->setAllowedTypes('query', 'string')
+        ->setAllowedTypes('limit', 'int')
+        ->setAllowedValues('limit', range(1, 1000));
+
+        $options = $resolver->resolve($options);
+
+        $endpoint = Endpoint::bind(Endpoint::GUILD_MEMBERS . '/search', $this->id);
```

Comment:
Could add in the `DiscordPHP-Http` library as well? I'll merge and can be done later